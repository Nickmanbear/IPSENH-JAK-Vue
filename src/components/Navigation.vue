<template>
  <div id="nav">
    <router-link to="/">Board Overview</router-link>
    <router-link v-if="authenticated === true" to="/logout">Logout</router-link>
    <router-link v-if="authenticated === false" to="/login">Login</router-link>
    <router-link v-if="authenticated === false" to="/register">Register</router-link>


  </div>
</template>

<script>

export default {
  name: 'Navigation',
  data() {
    return {
      authenticated: false,
    };
  },
  mounted() {
    this.getAuthenticated();
  },
  methods: {
    getAuthenticated() {
      this.authenticated = this.$store.getters.isLoggedIn;
    },
  },

};
</script>

<style lang="scss">
  #nav {
    background-color: #cccccc;
    width: calc(100% - 16px);
    padding: 8px;

    a {
      font-weight: bold;
      color: #2c3e50;
      margin: 16px;

      &.router-link-exact-active {
        color: #42b983;
      }
    }
  }
</style>
