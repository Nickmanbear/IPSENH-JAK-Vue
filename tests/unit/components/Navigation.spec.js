import { shallowMount } from '@vue/test-utils';

import Vue from 'vue';
import Vuex from 'vuex';
import Navigation from '@/components/Navigation.vue';


Vue.use(Vuex);

describe('Navigation', () => {
  let cmp;
  let store;
  let state;
  let getters;

  beforeEach(async () => {
    state = { token: false };
    getters = {
      isLoggedIn: (state) => false,
    };

    store = new Vuex.Store({
      state,
      getters,
    });
    cmp = await shallowMount(Navigation, {
      stubs: ['router-link'],
      data() {
        return {
          authenticated: false,
        };
      },
      store,
    });
  });

  it('should set Authenticated', async () => {
    expect(getters.isLoggedIn()).toEqual(false);
  });
  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot();
  });
});
