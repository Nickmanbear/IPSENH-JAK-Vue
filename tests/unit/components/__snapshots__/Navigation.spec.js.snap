// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation has the expected html structure 1`] = `
<div
  id="nav"
>
  <router-link-stub
    to="/"
  >
    <img
      alt="logo"
      src="undefinedimages/jak.png"
    />
  </router-link-stub>
   
  <div>
    <router-link-stub
      to="/login"
    >
      Login
    </router-link-stub>
     
    <router-link-stub
      to="/register"
    >
      Register
    </router-link-stub>
  </div>
</div>
`;
